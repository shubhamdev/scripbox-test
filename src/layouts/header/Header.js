import React from "react";

function Header(props) {
  return <div className="header">header</div>;
}

export default Header;
