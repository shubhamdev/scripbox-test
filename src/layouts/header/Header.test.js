import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { shallow } from "enzyme";
import Header from "./Header";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Header", () => {
  it('should render Header correctly in "debug" mode', () => {
    const component = shallow(<Header debug />);
    expect(component).toMatchSnapshot();
  });
  it("should render Header correctly with no props", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
