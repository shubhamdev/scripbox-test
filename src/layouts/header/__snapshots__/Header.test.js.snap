// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly in "debug" mode 1`] = `
<div
  className="header"
>
  header
</div>
`;

exports[`Header should render Header correctly with no props 1`] = `
<div
  className="header"
>
  header
</div>
`;
